extensions/sql-database-projects/src/dialogs/addSqlBindingQuickpick.ts

```diff
@@ -150,29 +152,73 @@ export async function launchAddSqlBindingQuickpick(uri: vscode.Uri | undefined,
 					// go back to select setting quickpick if user escapes from inputting the setting name in case they changed their mind
 					continue;
 				}
-
-				const newConnectionStringValue = await vscode.window.showInputBox(
+				let connectionStringOptions: vscode.QuickPickItem[] = [
 					{
-						title: constants.enterConnectionString,
-						ignoreFocusOut: true,
-						value: 'Server=localhost;Initial Catalog={db_name};User ID=sa;Password={your_password};Persist Security Info=False',
-						validateInput: input => input ? undefined : constants.valueMustNotBeEmpty
+						label: constants.connectionProfile
+					},
+					{
+						label: constants.userConnectionString
 					}
-				) ?? '';
-
-				if (!newConnectionStringValue) {
-					// go back to select setting quickpick if user escapes from inputting the value in case they changed their mind
-					continue;
+				];
+				const selectedConnectionString = await vscode.window.showQuickPick(connectionStringOptions, {
+					canPickMany: false,
+					title: constants.selectConnectionString,
+					ignoreFocusOut: true
+				});
+				if (!selectedConnectionString) {
+					// User cancelled
+					return;
 				}
-
-				try {
-					const success = await azureFunctionsUtils.setLocalAppSetting(path.dirname(projectUri.fsPath), newConnectionStringSettingName, newConnectionStringValue);
-					if (success) {
-						connectionStringSettingName = newConnectionStringSettingName;
+				if (selectedConnectionString.label === constants.userConnectionString) {
+					const newConnectionStringValue = await vscode.window.showInputBox(
+						{
+							title: constants.enterConnectionString,
+							ignoreFocusOut: true,
+							value: 'Server=localhost;Initial Catalog={db_name};User ID=sa;Password={your_password};Persist Security Info=False',
+							validateInput: input => input ? undefined : constants.valueMustNotBeEmpty
+						}
+					) ?? '';
+					if (!newConnectionStringValue) {
+						// go back to select setting quickpick if user escapes from inputting the value in case they changed their mind
+						continue;
+					} else {
+						continue;
+					}
+				} else {
+					// get connection string for sql server
+					let connectionString: string = '';
+					let connectionUri: string = '';
+					const connectionInfo = await vscodeMssqlApi.promptForConnection(true);
+					if (!connectionInfo) {
+						// User cancelled
+						return;
+					}
+					try {
+						// TO DO: https://github.com/microsoft/azuredatastudio/issues/18012
+						connectionUri = await vscodeMssqlApi.connect(connectionInfo);
+					} catch (e) {
+						// give an error if unable to connect to selected connection and go back to select setting quickpick
+						console.warn(e);
+						void vscode.window.showErrorMessage(utils.getErrorMessage(e));
```

Comment:
What's the error message here look like? Might be good to add a prefix text like `Connection to server failed: <ERROR>`

Reply:
I will add the prefix as it makes sense since they failed at that step. Here is what the error looks like: I will add the prefix as it makes sense since they failed at that step. Here is what the error looks like: 
![Screen Shot 2022-01-10 at 3 47 54 PM](https://user-images.githubusercontent.com/23587151/148857351-a97dbcae-cdf0-4b48-9724-0dd32cc482d9.png)

Reply:
Discussed this with @chlafreniere and since mssql sends out an error on the connection I just delete the one that I added in order to not have multiple warnings sent to the user. Potentially could enhance the mssql ext to include a silent failure to connect() API in order to add our specific warning instead.